src/point_one/fusion_engine/parsers/fusion_engine_framer.cc

```diff
@@ -253,37 +253,67 @@ int32_t FusionEngineFramer::OnByte(bool quiet) {
 
     // Check if the header is complete.
     if (next_byte_index_ == sizeof(MessageHeader)) {
-      // Compute the full message size. If the message is too large to fit in
-      // the buffer, we cannot parse it. Otherwise, start collecting the
-      // message payload.
-      //
-      // Note that while we compute the current_message_size_ here, we
-      // intentionally do the "too big" check below with the payload size. That
-      // way we implicitly handle cases where the payload is large enough to
-      // cause current_message_size_ to overflow. Normally, this won't happen
-      // for legit packets that are just too big for the user's buffer, but it
-      // could happen on a bogus header if we find the preamble randomly in an
-      // incoming byte stream. The buffer capacity is always
-      // >=sizeof(MessageHeader), so the subtraction will never be negative.
       auto* header = reinterpret_cast<MessageHeader*>(buffer_);
       current_message_size_ =
           sizeof(MessageHeader) + header->payload_size_bytes;
       VLOG(3) << "Header complete. Waiting for payload. [message="
-              << header->message_type << " (" << (unsigned)header->message_type
-              << "), seq=" << header->sequence_number
-              << ", payload_size=" << header->payload_size_bytes << " B]";
-      if (header->payload_size_bytes <=
-          capacity_bytes_ - sizeof(MessageHeader)) {
-        // If there's no payload, do the CRC check now.
-        if (header->payload_size_bytes == 0) {
-          VLOG(3) << "Message has no payload. Checking CRC.";
-          crc_check_needed = true;
+              << header->message_type << ", seq=" << header->sequence_number
+              << ", payload_size=" << header->payload_size_bytes
+              << " B, message_size=" << current_message_size_ << " B]";
+
+      // Check for size overflow. We don't currently expect to have _extremely_
+      // large packets, so this should never happen. If it does, assume this is
+      // not a valid message, and instead the sync pattern likely showed up at
+      // random in the data stream.
+      if (current_message_size_ < header->payload_size_bytes) {
```

Comment:
This check appears to be wrong. Shouldn't it be `current_message_size_ < capacity_bytes_ `?

Reply:
No, this one is intentional. It's checking for overflow of the uint32. The check for capacity is done below.

Reply:
Oh I see this is to check for an u32 overflow. Never mind.

Reply:
Though to be fair this seems like this would be simpler to have a MAX_SIZE you're checking that could be MAX_INTEGER-sizeof(MessageHeader) if you really wanted. I guess I'm not sure why you rewrote this check since the previous formulation was overflow safe and also covers the capacity check.

Reply:
Yeah, I changed it mainly to make things more consistent between this and the Python code, which doesn't use uint32. It adds more lines of code but I think it also makes it a little more clear since the old one was really checking for 2 independent conditions subtly (though with a note). I'll definitely call out that this is about uint32 rollover.